bgflow/nn/flow/transformer/jax_bridge.py

```diff
@@ -118,31 +125,123 @@ def bijector_with_approx_inverse(bijector, domain=None):
     """Wraps bijector with approximate inverse."""
     if domain is None:
         domain = (0, 1)
-
     root_finder = functools.partial(
         bisect,
         left_bound=domain[0],
         right_bound=domain[1])
-    invert_ = functools.partial(
-        invert,
+    invert = functools.partial(
+        invert_bijector,
         root_finder=root_finder)
-    bijector = with_ldj(jax.vmap(bijector))
+    forward = with_ldj(bijector)
+    inverse = invert(forward)
+    return forward, inverse
 
-    def _forward(x, cond):
-        return bijector(x, *cond)
 
-    def _inverse(x, cond):
-        return invert_(_forward)(x, cond)
+def assert_contiguous(x, params):
+    if not x.is_contiguous():
+        x = x.contiguous()
+    params = tuple(
+        p.contiguous() if not p.is_contiguous() else p
+        for p in params)
+    return x, params
 
-    return _forward, _inverse
+
+def assert_float32(x):
+    if x.dtype != torch.float32:
+        raise ValueError(f'dtype {x.dtype} currently not supported by jax bridge')
+        # if we want to support this, enable x64 globally during compile via a context mgr such as:
+        # >>> from jax.config import config
+        # >>> config.update("jax_enable_x64", True)
+        # >>> yield
+        # >>> config.update("jax_enable_x64", False)
+
+
+def nested_vmap(fn, indices):
+    """Applies nested vmap for specified vectorization indices."""
+    for idx in indices:
+        fn = jax.vmap(fn, in_axes=idx, out_axes=idx)
+    return fn
+
+
+def wrap_params(fn):
+    """Makes signature compatible with jax2torch."""
+    def _call(x, params):
+        return fn(x, *params)
+    return _call
 
 
-def to_bgflow(bijector, domain=None):
+def jax_compile(bijector, vmap_indices, backend, domain=None,):
     """Wraps simple JAX bijector into a transformer,
        that can be used within the bgflow eco-system."""
-    return map(compose(
-        functools.wraps(bijector),
-        jax2torch.jax2torch,
-        jax.jit,
-        jax.vmap),
-        bijector_with_approx_inverse(bijector, domain))
+    compile_bijector = compose(functools.partial(jax.jit, backend=backend), wrap_params)
```

Comment:
nice! 
Now that you have a dedicated function and caching for this, it should be easy to support double precision, too.

Reply:
Very well possible :)